modulemd/v2/modulemd-service-level.c

```diff
@@ -55,6 +55,30 @@ modulemd_service_level_new (const gchar *name)
 }
 
 
+gboolean
+modulemd_service_level_equals (ModulemdServiceLevel *self_1,
+                               ModulemdServiceLevel *self_2)
+{
+  g_return_val_if_fail (MODULEMD_IS_SERVICE_LEVEL (self_1), FALSE);
+  g_return_val_if_fail (MODULEMD_IS_SERVICE_LEVEL (self_2), FALSE);
+
+  if (g_strcmp0 (modulemd_service_level_get_name (self_1),
+                 modulemd_service_level_get_name (self_2)) != 0)
+    return FALSE;
+
+  if ((!self_1->eol && self_2->eol) || (!self_2->eol && self_1->eol))
+    return FALSE;
+
+  if (!g_date_valid (self_1->eol) || !g_date_valid (self_2->eol))
```

Comment:
`g_date_valid()` will crash if passed a `NULL` value, which is possible if both `self_1->eol` and `self_2->eol` are `NULL` above. I think it's best if we just document that this function will always return false if either service level passed in contains a NULL or invalid date, since they cannot be legitimately compared. If they are *both* NULL, we can treat that as a match and return that immediately (and skip the `g_date_valid()` and `g_date_compare()` calls)